src/features/twitter/controllers/twitter-controller.test.ts

```diff
@@ -69,9 +65,25 @@ describe('[Controller] Twitter', () => {
   });
 
   describe('login', () => {
-    it('should be return 401', () => {
-      const { authController, next, req, res } = makeSut();
+    it('return URLs when the token is valid', async () => {
+      const mockPayload = {
+        name: 'John Doe',
+        userId: '12345',
+        username: 'johndoe',
+      };
+      const expectedUrl = `https://twitter.com/i/oauth2/authorize?client_id=mockClientId123&code_challenge=-a4-ROPIVaUBVj1qqB2O6eN_qSC0WvET0EdUEhSFqrI&code_challenge_method=S256&redirect_uri=http%3A%2F%2Fwww.localhost%3A3000%2Fapi%2Ftwitter%2Fcallback&response_type=code&state=${mockPayload.userId}&scope=tweet.write%20tweet.read%20users.read`;
+      process.env.TWITTER_CLIENT_ID = 'mockClientId123';
+      req.headers.authorization =
+        'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiSm9obiBEb2UiLCJ1c2VySWQiOiIxMjM0NSIsInVzZXJuYW1lIjoiam9obmRvZSIsImlhdCI6MTY2NTMzMjM0NCwiZXhwIjoxNjY1MzM1OTQ0fQ.S5ReMBrQqVAD6UCD6Q9Vj9fK9J-Q9r_a9f3zRmpDsxM';
+
+      vi.spyOn(jwt, 'verify').mockReturnValue(mockPayload as any);
+
+      await authController.login(req, res, next);
+
+      expect(res.json).toHaveBeenCalledWith(expectedUrl);
+    });
 
+    it('returns 401 when token is invalid', () => {
       req.headers.authorization = undefined;
 
       authController.login(req, res, next);
```

Comment:
aqui não falta um await pra ter consistencia?
```suggestion
     await authController.login(req, res, next);
```

Reply:
removi o outro await